app/src/main/res/layout/item_homework.xml

```diff
@@ -0,0 +1,146 @@
+                            app:layout_constraintBottom_toBottomOf="parent"
+                            app:layout_constraintEnd_toEndOf="parent"
+                            app:layout_constraintHorizontal_bias="0.0"
+                            app:layout_constraintStart_toStartOf="parent"
+                            app:layout_constraintTop_toTopOf="parent"
+                            tools:text="Course" />
+
+                        <ImageView
+                            android:layout_width="wrap_content"
+                            android:layout_height="wrap_content"
+                            android:paddingTop="1dp"
+                            android:src="@drawable/circle_course_color"
+                            app:setcolor="@{Color.parseColor(homework.courseId.color)}"
+                            app:layout_constraintStart_toEndOf="@+id/course_title"
+                            app:layout_constraintTop_toTopOf="parent" />
+
+                    </android.support.constraint.ConstraintLayout>
+
+                    <ImageView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:visibility="@{ViewUtilsKt.asVisibility(HomeworkViewHolder.dueLabelFlagRequired(homework))}"
+                        app:srcCompat="@drawable/baseline_outlined_flag_16dp" />
+
+                    <TextView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_weight="0"
+                        android:textColor="@{safeUnbox(homework.dueTextAndColorId.second)}"
+                        android:textSize="@dimen/text_smaller_size"
+                        android:text="@{homework.dueTextAndColorId.first}"
+                        tools:text="In x Stunden fällig" />
+
+                </LinearLayout>
+
+                <TextView
+                    style="@style/TextAppearance.AppCompat.Medium"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="5dp"
+                    android:gravity="start"
+                    android:textColor="@android:color/black"
+                    android:textStyle="bold"
+                    android:text="@{homework.title}"
+                    tools:text="Homework Title" />
+
+                <TextView
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="8dp"
+                    android:ellipsize="end"
+                    android:gravity="start"
+                    android:maxLines="4"
+                    android:textSize="@dimen/text_medium_size"
+                    android:text="@{HomeworkViewHolder.getTextFromHtml(homework.description)}"
```

Comment:
For showing HTML, please use either ContentWebView or PassiveWebView (the latter in this case). TextView doesn't support all HTML features. Also the HTML-parsing blocks the UI thread which can cause dropped frames. WebView takes care of this!

Reply:
For the homework preview I need to limit the number of lines displayed, as well as adding an ellipsize at the end. WebView doesn't support this.